src/spikeinterface/preprocessing/deepinterpolation/generators.py

```diff
@@ -0,0 +1,311 @@
+                        (
+                            total_samples_pre + num_samples_segment_pre - pre_extended - 1,
+                            total_samples_pre + num_samples_segment_pre + post_extended,
+                        )
+                    )
+            # exclude samples at the border of the recordings
+            if total_samples_pre > 0:
+                exclude_intervals.append((total_samples_pre - pre_extended - 1, total_samples_pre + post_extended))
+
+        total_valid_samples = (
+            total_num_samples - np.sum([end - start for start, end in exclude_intervals]) - pre_extended - post_extended
+        )
+        self.total_samples = int(total_valid_samples) if total_samples == -1 else total_samples
+        assert len(desired_shape) == 2, "desired_shape should be 2D"
+        assert (
+            desired_shape[0] * desired_shape[1] == recording.get_num_channels()
+        ), f"The product of desired_shape dimensions should be the number of channels: {recording.get_num_channels()}"
+        self.desired_shape = desired_shape
+
+        start_frame = start_frame if start_frame is not None else 0
+        end_frame = end_frame if end_frame is not None else total_num_samples
+        assert end_frame > start_frame, "end_frame must be greater than start_frame"
+
+        sequential_generator_params = dict()
+        sequential_generator_params["steps_per_epoch"] = steps_per_epoch
+        sequential_generator_params["pre_frame"] = pre_frame
+        sequential_generator_params["post_frame"] = post_frame
+        sequential_generator_params["batch_size"] = batch_size
+        sequential_generator_params["start_frame"] = start_frame
+        sequential_generator_params["end_frame"] = end_frame
+        sequential_generator_params["total_samples"] = self.total_samples
+        sequential_generator_params["pre_post_omission"] = pre_post_omission
+
+        json_path = tempfile.mktemp(suffix=".json")
+        with open(json_path, "w") as f:
+            json.dump(sequential_generator_params, f)
+        super().__init__(json_path)
+
+        self._update_end_frame(total_num_samples)
+
+        # self.list_samples will exclude the border intervals on the concat recording
+        self.recording_concat = concatenate_recordings(recordings)
+        self.exclude_intervals = exclude_intervals
+        self._calculate_list_samples(total_num_samples, exclude_intervals=exclude_intervals)
+
+        self._kwargs = dict(
+            recordings=recordings,
+            pre_frame=pre_frame,
+            post_frame=post_frame,
+            pre_post_omission=pre_post_omission,
+            desired_shape=desired_shape,
+            batch_size=batch_size,
+            steps_per_epoch=steps_per_epoch,
+            start_frame=start_frame,
+            end_frame=end_frame,
+        )
+
+    # this is overridden to exclude samples from borders
+    def _calculate_list_samples(self, total_frame_per_movie, exclude_intervals=[]):
```

Comment:
I wonder if this is needed given that you are inheriting from the SequentialGenerator

Reply:
in this case it's needed because of the borders exclusion